src/vm-repair/azext_vm_repair/_validators.py

```diff
@@ -201,6 +204,18 @@ def _prompt_repair_password(namespace):
         raise CLIError('Please specify the password parameter in non-interactive mode.')
 
 
+def _prompt_public_ip(namespace):
+    from knack.prompting import prompt_y_n, NoTTYException
+    try:
+        if prompt_y_n('Does repair vm requires public ip?'):
+            namespace.associate_public_ip = "yes"
+        else:
+            namespace.associate_public_ip = '""'
+
+    except NoTTYException:
+        raise CLIError('Please specify the associate-public-ip parameter in non-interactive mode.')
```

Comment:
Could you please use specific error type instead of `CLIError`. For more details, please refer to [error_handling_guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

Reply:
Specified specific error type **ValidationError** instead of **CLIError**

Below are the test cases.
**Test case 1 :**
Repair command can be executed without argument --associate-public-ip . Then command will wait for user input with below prompt.
 "Does repair vm requires public ip? (y/n)"
 a) y input should create repair vm with public ipaddress
 b) n input should create repair VM without public ipaddress
 
**Test case 2 :**
Repair command can be executed with argument--associate-public-ip . User won’t be prompted for public IP option instead repair VM will be created with Public IP associated to it.